### PR TITLE
Fixed #3363 - Export Target List fails (Database failure)

### LINKED ISSUE #3363: Export Target List fails (Database failure)
#### Issue

I get a database error when trying to export the target list as csv as soon as any of the related records include a custom HTML field.

#### Expected Behavior

Target list export works no matter what custom fields are added to modules contacts, leads, targets and accounts.

#### Actual Behavior

Target list export fails when any of the linked modules have a custom HTML field set up.

#### Possible Fix

Reason for the bug:
The mysql query being fired upon export action treats the static HTML field type the same as normal module fields, i.e. it's looking for a column in the ".cstm" table for the respective module. For field type HTML (and perhaps others?) there is no column in the ".cstm" field, so mysql throws an error.

#### Steps to Reproduce

1. set up a custom field of type HTML in one of the four target-list related modules
2. add a record of respective module to a target list
3. export target list as csv (from the actions menu in target list detail view)
4. -> Database error

#### Your Environment

* SuiteCRM Version used: 7.7.7
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)):
* Environment name and version (e.g. MySQL, PHP 7):
* Operating System and version (e.g Ubuntu 16.04):

### PR DESCRIPTION
**Pull request by ChangezKhan**
It fixes #3363 

## Description

Added condition to check if the field is of non-db type or not.

Steps to Reproduce

1.   Set up a custom field of type HTML in one of the four target-list related modules
2.   Add a record of respective module to a target list
3.   Export target list as csv (from the actions menu in target list detail view)
4.   Database error

## Motivation and Context

Target list export fails when any of the linked modules have a custom HTML(non-db) field set up. 

## How To Test This

Steps to Reproduce

1.   Set up a custom field of type HTML in one of the four target-list related modules
2.   Add a record of respective module to a target list
3.   Export target list as csv (from the actions menu in target list detail view)
4.   Database error

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: modules/ProspectLists/ProspectList.php [hunk 1]
```diff
@@ -204,8 +206,11 @@ function create_export_members_query($record_id)
 
 			foreach($members as $membername => &$memberarr)
 			{
+				$module_name = $beanList[$membername];
+				require_once($beanFiles[$module_name]);
+				$relatedBean = BeanFactory::getBean($membername);
 				// if the field belongs to this module, then query it in the cstm table
-				if ($membername == $val['custom_module'])
+				if ($membername == $val['custom_module'] && $relatedBean->field_defs[$val['name']]['source'] != 'non-db')
```
Comment (daniel-samson) [inline modules/ProspectLists/ProspectList.php]: PSR-2 Safely use !==. Can you please change the != to !==?

Comment (daniel-samson) [inline modules/ProspectLists/ProspectList.php]: Can you please change == to === ?

**Comment by Dillon-Brown** [comment]
@daniel-samson Works as expected

**Comment by daniel-samson** [comment]
@samus-aran can you please merge?

**Pull request merged**